sys/riotboot/flashwrite.c

```diff
@@ -103,15 +109,19 @@ int riotboot_flashwrite_putbytes(riotboot_flashwrite_t *state,
     LOG_DEBUG(LOG_PREFIX "processing bytes %u-%u\n", state->offset, state->offset + len - 1);
 
     while (len) {
-        size_t flashpage_pos = state->offset % FLASHPAGE_SIZE;
+        /* Position within the page */
+        size_t flashpage_pos = state->offset -
+            (flashpage_addr(state->flashpage) - (void*)riotboot_slot_get_hdr(state->target_slot));
```

Comment:
I had to think about this for a bit, but it makes sense.
Wouldn't hurt expanding the comment a bit.

Reply:
Clarified!